src/test/java/org/jabref/logic/bibtex/comparator/EntryComparatorTest.java

```diff
@@ -14,4 +15,70 @@ void recognizeIdenticObjectsAsEqual() {
         BibEntry e2 = e1;
         assertEquals(0, new EntryComparator(false, false, StandardField.TITLE).compare(e1, e2));
     }
+
+    /**
+     * Test if the compare function correctly sorts two BibTex entries with
+     * author names only having one differing letter.
+     */
+    @Test
+    void testEntryComparator1() {
+        BibEntry e1 = new BibEntry(StandardEntryType.Article);        
+        e1.setField(StandardField.AUTHOR, "ABB");        
+        
+        BibEntry e2 = new BibEntry(StandardEntryType.Article);        
+        e2.setField(StandardField.AUTHOR, "ABC");        
+
+        assertEquals(-1, new EntryComparator(false, false, StandardField.AUTHOR).compare(e1, e2));
+        assertEquals(1, new EntryComparator(false, false, StandardField.AUTHOR).compare(e2, e1));
+    }
+    
+    /**
+     * Test if compare function correctly sorts two BibTex entries in
+     * regards to the surname of the author
+     */
+    @Test
+    void testEntryComparator2() {
+        BibEntry e1 = new BibEntry(StandardEntryType.Article);        
+        e1.setField(StandardField.AUTHOR, "ABC ABB");        
+        
+        BibEntry e2 = new BibEntry(StandardEntryType.Article);        
+        e2.setField(StandardField.AUTHOR, "ABB ABC");        
+
+        assertEquals(1, new EntryComparator(false, false, StandardField.AUTHOR).compare(e1, e2));
```

Comment:
Does 1 mean that `e1 > e2`? What is the surname in this case, is it `e1->ABB` and `e2->ABC`? I might've misunderstood something here!

Reply:
Looking into it more, I think the function interprets the first word as the surname. As if the author's name would be written as "Knuth, Donald" which results in `e1 > e2`.